SUS-1278 Fix issues found in Special:PhalanxStats for wiki level stats

garthwebb opened the pull request:
A change in a recent PR for this component changed a template which broke the display of Wiki level Phalanx stats.  This PR cleans up some of the controller code and adds a new template specifically for this data.  More info at:

https://wikia-inc.atlassian.net/browse/SUS-1278

Also, note that the branch name is incorrect.  It should be **SUS** not **SOC**, but its already created.

@nmonterroso

Commit:
SOC-1278 Fix issues found in Special:PhalanxStats for wiki level stats

WikiaTech commented on extensions/wikia/PhalanxII/classes/PhalanxStatsWikiaPager.class.php:
```diff
@@ -10,20 +12,44 @@ public function __construct( $id ) {
+			[ 'parseinline', 'replaceafter' ],
+			$type, $username, $phalanxUrl, $timestamp, $statsUrl, $url
+		);
 		$html .= Html::closeElement( 'li' );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,118 @@ function __construct( ) {
+
+		// pull these out of the array, so they don't get used in the top rows
+		$row = $data->toArray();
+		unset( $row['text'], $row['reason'], $row['comment'], $row['ip_hex'] );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reason" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "comment" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,118 @@ function __construct( ) {
 	}
 
 	public function index() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

garthwebb replied:
Made a few updates in this function to reduce indentation and also make sure to return True/False from the method since returning false is necessary to inhibit the template from generating when there is no content or an error.

WikiaTech commented:
SonarQube analysis reported 11 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 3 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L235](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L235): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxStatsSpecialController.class.php#L132](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php#L132): Define a constant instead of duplicating this literal "expire" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L178](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L178): Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L191](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L191): Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L235](https://github.com/Wikia/app/blob/2b50e6d2ab932c4bbeeafff1cfc00ab8bd97ea0a/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L235): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,118 @@ function __construct( ) {
 
 	public function index() {
-		wfProfileIn( __METHOD__ );
 		$this->wg->Out->setPageTitle( wfMsg( 'phalanx-stats-title' ) );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "phalanx-stats-title" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

garthwebb commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -66,109 +139,59 @@ private function blockStats( $blockId ) {
+		return $content;
 	}
 
-	private function blockWikia( $wikiId ) {
```
Split this method into several smaller methods.

nmonterroso commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,118 @@ function __construct( ) {
+			'comment' => $this->getParsedContent( $data, 'comment' ),
+			'editUrl' => $this->phalanxTitle->getLocalURL( [ 'id' => $data['id'] ] ),
+			'blockId' => $blockId,
+		    'statsPager' => $this->buildPager( $blockId )
```
I think the indentation is off here

nmonterroso reviewed (approved):
minor comment about indentation otw lgtm 👍

Commit:
SUS-1278 Fixed indentation

WikiaTech commented:
SonarQube analysis reported 11 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 3 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L235](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L235): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxStatsSpecialController.class.php#L132](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php#L132): Define a constant instead of duplicating this literal "expire" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L178](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L178): Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L191](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L191): Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L235](https://github.com/Wikia/app/blob/667d4a78d084ee85f496a4531364a2f2d6cc4a1e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L235): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

Commit:
SUS-1278 Make the getTypeNames a little more understandable and allow for bad data

WikiaTech commented:
SonarQube analysis reported 27 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 19 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/673d028da8cc028c55bec5419f413d025a87ec23/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

Commit:
Merge branch 'dev' into SOC-1278

WikiaTech commented:
SonarQube analysis reported 27 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 19 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/a94618d256ca77cc4ae95ae7353f9c40c65379d6/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

Commit:
SUS-1215 Be more forgiving of older blocks

Older blocks may not have the ps_blocker_hit column set in the phalanx_stats table. Existing code was using the null coalescing operator to check both ps_blocker_hit then ps_blocker_type.  However as this value is sometimes zero as well as null, this did not do the right thing. Also add back in an old block for stats display purposes, but hide it for block creation.

Commit:
SUS-1215 Don't create a URL if the server is not present.

The GlobalTitle will create a URL even the server name is not available, leading to a broken URL.

Also update the schema to reflect reality.

Commit:
SUS-1215 Update wfMsgExt to wfMessasge(...)

garthwebb committed:
Merge pull request #11847 from Wikia/SUS-1215

SUS-1215 Be more forgiving of older blocks

WikiaTech commented:
SonarQube analysis reported 31 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 20 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/ce915f8b281e80e1b8af05f2056a3e090a7d6843/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -141,7 +183,12 @@ public function save() {
+				__METHOD__
+			);
 			$action = 'edit';
 		};
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)

nmonterroso commented:
👍

mszabo-wikia commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,118 @@ function __construct( ) {
+	 *
+	 * @return bool
+	 */
+	public function blockWiki() {
```
Let's add access control to this function. All Nirvana controllers' public methods are accessible via the API so we need to check if user is allowed to use Phalanx to avoid information leaking.

We could implement `userAllowedRequirementCheck` method which is automatically called and checked on the dispatcher level so that we don't have to repeat this check for every method.

mixth-sense replied:
I think we can live with the simple check:

```
if ( !$this->userCanExecute( $this->wg->User ) ) {↵
    $this->displayRestrictionError();↵
    return false;↵
}↵
```
With current (lack of) implementation ACLs, a) we still need to call `userCanExecute()` whenever we want to limit execution; b) we still need to `displayRestrictionError()` in `$this` context; c) we still need to `return false` to terminate exection. :-(

mszabo-wikia replied:
@mixth-sense But if we implement `userAllowedRequirementCheck` method we only need to `return !$user->isAllowed( 'phalanx' );` because dispatcher will abort execution:

https://github.com/Wikia/app/blob/dev/includes/wikia/nirvana/WikiaDispatcher.class.php#L210

Commit:
add a restriction check in blockWiki()

WikiaTech commented:
SonarQube analysis reported 31 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 20 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/d23c26ab808a06f54bf39a56d6629b0bec7b27cc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

macbre commented on extensions/wikia/PhalanxII/templates/PhalanxStatsSpecial_blockWiki.php:
```diff
@@ -0,0 +1,19 @@
+	</tr>
+	<tr>
+		<td><?= $wikiData['wiki_id'] ?></td>
+		<td><?= $wikiData['sitename'] ?></td>
```
Let's add HTML encoding here

macbre commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,124 @@ function __construct( ) {
+			'firstRow' => $row,
+			'text' => $data['text'],
+			'reason' => $this->getParsedContent( $data, 'reason' ),
+			'comment' => $this->getParsedContent( $data, 'comment' ),
```
Can we use `Linker::formatComment` instead? It's a way lighter version of wikitext parser. I don't think that we need to handle templates or images here - pinging @Kirkburn to be sure :smile:

Kirkburn replied:
Yeah, we don't need to handle complex wikitext content in block comments - only links.
